### PR TITLE
Fix Hosted AMP inline videos

### PR DESCRIPTION
**Pull request by lps88**
## What does this change?
Convert inline videos in Hosted AMP article pages to `<amp-video>` tags, and remove the `<gu-atom>` tag which is invalid in AMP. 

## What is the value of this and can you measure success?
Inline videos now work, where they were previously being removed.

## Does this affect other platforms - Amp, Apps, etc?
Just AMP.

## Screenshots
image

## Request for comment
@guardian/labs-beta

### CODE REVIEW

File: commercial/app/model/hosted/HostedAmp.scala [hunk 1]
```diff
@@ -19,12 +19,24 @@ object HostedAmp {
...
     }
 
+    def transformVideos(): Unit = {
+      def transformVideo(video: Element): Element = video.tagName("amp-video").attr("layout", "responsive").attr("width", "16").attr("height", "9")
+      val figures = doc.select("figure:has(gu-atom)")
+      for (figure <- figures.asScala;
+           video <- figure.select("video").asScala) {
+
+        figure.appendChild(transformVideo(video))
+        figure.select("gu-atom").remove()
```
Comment (kelvin-chappell) [inline commercial/app/model/hosted/HostedAmp.scala]: Isn't this already being done by `removeAtoms()`?

**Comment by Calanthe** [comment]
Did you run the AMP test in Scala? Shouldn't you add some test fixtures?

**Review by kelvin-chappell — APPROVED** [review]
Apart from some possible duplication 👍

File: commercial/app/model/hosted/HostedAmp.scala (hunk 1 above)
Comment (lps88) [inline commercial/app/model/hosted/HostedAmp.scala]: The `removeAtoms` function removes the whole `figure` tag (and the `amp-video` inside it). I looked at modifying that function but seemed cleaner to keep it a separate operation.

**Comment by Calanthe** [comment]
👍

File: commercial/app/model/hosted/HostedAmp.scala (hunk 1 above)
Comment (Aliihi) [inline commercial/app/model/hosted/HostedAmp.scala]: https://github.com/guardian/frontend/commit/c878df6

**Comment by Aliihi** [comment]
https://avatars2.githubusercontent.com/u/24212887?v=3&s=400

**Pull request merged**